docs/concepts/typings.md

```diff
@@ -4,4 +4,103 @@ title: Type hints
+
+When using Strawberry to build graphQL APIs, as was shown in [Schema basics](https://strawberry.rocks/docs/general/schema-basics), type hints are required within classes decorated by `@strawberry.type` & `@strawberry.input` and functions decorated by `@strawberry.field` & `strawberry.mutation`. These type hints are sourced as the keywords `str`, `int`, `float`, and from packages imported directly from the Python standard libraries `typing`, that has been available since Python 3.5, `datetime`, and `decimal`.
+
+## Mapping to graphQL types
+
+The complete mapping of the required type hints for the relevant graphQL types is as follows:
+
+| GraphQL       | Python                     |
+| ------------- | -------------------------- | --- |
+| `ID`          | `strawberry.ID`            |
+| `String`      | `str`                      |
+| `Integer`     | `int`                      |
+| `Float`       | `float`                    |
+| `Decimal`     | `decimal.Decimal`          |
+| `Array`, `[]` | `typing.List` or `list`    |
+| `Union`       | `typing.Union` or `        | `   |
+| `Nullable`    | `typing.Optional` or `None | `   |
+| `date`        | `datetime.date`            |
+| `timetz`      | `datetime.time`            |
+| `timestamptz` | `datetime.datetime`        |
+
+where `typing`, `datetime`, and `decimal` are all part of the Python standard library. There is also `typing.Dict` that possesses no mapping since it is the entire structure of the graphQL query itself that is a dictionary.
+
+There are a few different ways in which these Python type hints can be used to express the required Strawberry graphQL type annotation.
+
+- For versions of Python >= 3.10, it is possible to annotate an array of types with `list[Type]`. However, for all previous versions, `typing.List[Type]` must be used instead.
+- The annotation `|` is shorthand for `typing.Union[]`, allowing either of `typing.Union[TypeA, TypeB]` or `TypeA | TypeB` interchangably.
+- The annotation `typing.Optional[Type]` is shorthand for `typing.Union[None, Type]`, which is itself equivalent to `None | Type`.
+
+## Example
+
+A complete example of this, extending upon [Schema basics](https://strawberry.rocks/docs/general/schema-basics), might be the following:
+
+```python
+import datetime
+import typing
+import strawberry
+
+BOOKS_LOOKUP = {
+    'Frank Herbert': [{
+        'title': 'Dune',
+        'date_published': '1965-08-01',
+        'price': '5.99',
+        'isbn': 9780801950773
+    }],
+}
+
+@strawberry.type
+class Book:
+    title: str
+    author: 'Author'
+    date_published: datetime.date
+    price: decimal.Decimal
+    isbn: Union[int, str] # could be 1234567890000 or '123-4-56-789000-0'
+
+def get_books_by_author(root: 'Author') -> typing.List['Book']:
```

Comment:
I think usually in other examples we've been importing _from_ the `typing` namespace and not using the package directly. We should be/continue to be consistent.

Reply:
Also, we may want to use `future` annotations

Reply:
I have changed the example to use Python 3.10 hinting style inline with @patrick91's comments, so there are no imports from typing